### PR TITLE
Updates codebase from commonjs to es-modules syntax

### PR DESCRIPTION
**Pull request by corocoto**
## Description
This PR should resolve the "TypeError: BpmnModdle is not a constructor" error. And should be consistent solution as for Node.js as for browser environment.

## Related PRs:
* https://github.com/bpmn-io/bpmn-auto-layout/pull/30
* https://github.com/bpmn-io/bpmn-auto-layout/pull/31

## Related issues:
* https://github.com/bpmn-io/bpmn-auto-layout/issues/18
* https://github.com/bpmn-io/bpmn-auto-layout/issues/22

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +33,7 @@
   "devDependencies": {
     "eslint": "^6.7.2",
     "eslint-plugin-bpmn-io": "^0.10.0",
-    "mocha": "^6.2.2",
+    "mocha": "^10.2.0",
```
Comment (corocoto) [inline package.json:36]: Updating to the latest version was neccessary to correctly work with es-module syntax

**Comment by nikku** [comment]
Looks good to me. @marstamm assigning you as a reviewer as you planned to look into this anyway?

File: example.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-var AutoLayout = require('bpmn-auto-layout');
+import AutoLayout from 'bpmn-auto-layout';
+import { promises as fs } from 'fs';
```
Comment (nikku) [inline example.js:2]:
```suggestion
import fs from 'fs/promises';
```

File: example.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-var AutoLayout = require('bpmn-auto-layout');
+import AutoLayout from 'bpmn-auto-layout';
```
Comment (nikku) [inline example.js:1]:
```suggestion
import { AutoLayout } from 'bpmn-auto-layout';
```

File: index.js [hunk 4]
```diff
@@ -1 +1 @@
-module.exports = require('./lib/AutoLayout');
+export { default } from './lib/AutoLayout.js';
```
Comment (nikku) [inline index.js:1]:
```suggestion
export { default as AutoLayout } from './lib/AutoLayout.js';
```

**Review by nikku — APPROVED** [review]
Looks great. I'll follow up with some cosmetic changes (update to README and stuff). :tada:

**Comment by corocoto** [comment]
@nikku, cool! Thanks for your kind words! And thanks a lot for your help 😊

**Pull request merged**